### DISCUSSION TITLE
Using Azure Devops Agents

### DISCUSSION THREAD

**Discussion by AHuusom** [actions/virtual-environments#3643] [Image configuration]
Hi.

We've successfully created Windows 2019 and Ubuntu2004 build servers using your repository. This is really formidable work you are doing.

But.... we've noticed that when running the azure devops agents it seems they don't have access to the environment variables, so f.ex. our build pipelines can't use YARN and Java.

We've tried to create an extra user during provisioning of the servers and we made the devops agent run as this user.
When doing a printenv as this user all environment variables and paths are present, but when running in pipelines all environment variables are gone and paths are down to a bare minimum.

Does anyone have some guidance on how to fix this?

**Comment by annes1230** [comment]


**Reply by annes1230** [reply]
dfc
